MapboxNavigation/VisualInstructionComponent.swift

```diff
@@ -0,0 +1,40 @@
+import UIKit
+import SDWebImage
+import MapboxDirections
+
+extension VisualInstructionComponent {
+    
+    static let scale = UIScreen.main.scale
+    
+    func shieldKey() -> String {
+        return "\(imageURL!.absoluteString)-\(VisualInstructionComponent.scale)"
+    }
+    
+    func cachedShield(_ shieldKey: String) -> UIImage? {
+        return SDImageCache.shared().imageFromCache(forKey: shieldKey)
+    }
+    
+    func shieldImage(height: CGFloat, completion: @escaping (UIImage?) -> Void) {
+        guard let imageURL = imageURL else { return }
+        let shieldKey = self.shieldKey()
+        
+        if let cachedImage = self.cachedShield(shieldKey) {
+            completion(cachedImage)
+            return
+        }
+        
+            
+        SDWebImageDownloader.shared().downloadImage(with: imageURL, options: [], progress: nil, completed: { (image, data, error, successful) in
+            guard let imageData = data else { return }
```

Comment:
Nit: drop these two guards and use `image` instead of `data`.

Reply:
Don't we need to properly scale the image though?

Reply:
Yes. The current implementation does scale it appropriately with `UIImage(data:scale)`, however, the image is being initialized with data on the main thread which could cause jittering. SDWebImage already used the data to create a UIImage on a background thread so there is no need for us to do the same again.